pkg/tsdb/testdatasource/flight_path.go

```diff
@@ -0,0 +1,135 @@
+package testdatasource
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+)
+
+func (p *testDataPlugin) handleFlightPathScenario(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	resp := backend.NewQueryDataResponse()
+
+	for _, q := range req.Queries {
+		_, err := simplejson.NewJson(q.JSON)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query json: %v", err)
+		}
+
+		timeWalkerMs := q.TimeRange.From.UnixNano() / int64(time.Millisecond)
+		to := q.TimeRange.To.UnixNano() / int64(time.Millisecond)
+		stepMillis := q.Interval.Milliseconds()
+
+		cfg := newFlightConfig()
+
+		maxPoints := q.MaxDataPoints * 2
+		f := cfg.initFields()
+		for i := int64(0); i < maxPoints && timeWalkerMs < to; i++ {
+			t := time.Unix(timeWalkerMs/int64(1e+3), (timeWalkerMs%int64(1e+3))*int64(1e+6))
+			f.append(cfg.getNextPoint(t))
+
+			timeWalkerMs += stepMillis
+		}
+
+		// When close to now, link to the live streaming channel
+		if q.TimeRange.To.Add(time.Second * 2).After(time.Now()) {
```

Comment:
What do you think is it possible to do somehow explicitly in the future?

Reply:
yes -- we should likely parse/keep the original string request and use that, if "to=now" could trigger live